apps/ctaltracker/cta_l_tracker.star

```diff
@@ -0,0 +1,289 @@
+        width = 36,
+        child = render.Text(arrival["destination_name"], font = "CG-pixel-4x5-mono", height = 7),
+    )
+    arrival_in_text = render.Text(arrival["eta_text"], color = "#f3ab3f")
+
+    stack = render.Stack(children = [
+        background_color,
+    ])
+
+    column = render.Column(
+        children = [
+            destination_text,
+            arrival_in_text,
+        ],
+    )
+
+    return render.Row(
+        expanded = True,
+        main_align = "space_evenly",
+        cross_align = "center",
+        children = [
+            stack,
+            column,
+        ],
+    )
+
+"""
+Creates a Row and adds needed children objects
+for a single arrival.
+"""
+
+def get_selected_station_map_id(selected_station):
+    stations = get_stations()
+    if not stations:
+        return None
+
+    for station in stations:
+        if station["station_descriptive_name"] == selected_station:
+            return station["map_id"]
+    fail("The stop selected was not matched to a formatted stop")
+
+"""
+Gets a list of "L" stations from API and
+eliminates duplicate stations
+"""
+
+def get_stations():
+    cache_stations = cache.get(L_STOPS_CACHE_KEY)
+    if cache_stations != None:
+        return json.decode(cache_stations)
+
+    print("Miss! No L Stops info in cache, calling L Stops API.")
+
+    app_token = secret.decrypt(ENCRYPTED_L_STOPS_APP_TOKEN)
+    if not app_token:
+        return None
+
+    response = http.get(
+        CTA_STATIONS_URL,
+        params = {
+            "$$app_token": app_token,
+        },
+    )
+    if response.status_code != 200:
+        fail("CTA L Stops request failed with status %d", response.status_code)
+
+    data = response.json()
+    if len(data) == 0:
+        fail("CTA L Stops API returned no stops")
+
+    stations = [build_station(station) for station in data]
+    deduped_stations = [i for n, i in enumerate(stations) if i not in stations[n + 1:]]
+    cache.set(L_STOPS_CACHE_KEY, json.encode(deduped_stations), ttl_seconds = 3600)
+    return deduped_stations
+
+"""
+Formats list of "L" stations into options
+for dropdown
+"""
+
+def get_station_options(station_mapping):
+    if not station_mapping:
+        return None
+
+    station_options = [
+        schema.Option(display = "%s" % station["station_descriptive_name"], value = station["station_descriptive_name"])
+        for station in station_mapping
+    ]
+    return station_options
+
+"""
+Creates a dictionary for the passed in "L" station
+containing station name and id
+"""
+
+def build_station(station):
+    return {
+        "station_descriptive_name": station["station_descriptive_name"],
+        "map_id": station["map_id"],
+    }
+
+"""
+Gets top 2 arrivals scheduled for the selected station
+from CTA Arrivals API
+"""
+
+def get_journeys(station_code):
+    api_key = secret.decrypt(ENCRYPTED_ARRIVALS_API_KEY)
+    if api_key == None or station_code == None:
+        return None
+
+    response = http.get(
```

Comment:
Could you add some caching for this call?

Reply:
Sure, I'll add a 60 second cache to this